transport/messages/result_message_base.hh

```diff
@@ -49,14 +49,15 @@ public:
     void add_tablet_info(locator::tablet_replica_set tablet_replicas, std::pair<dht::token, dht::token> token_range) {
         if (!tablet_replicas.empty()) {
             auto replicas_values = make_list_value(replica::get_replica_set_type(), replica::replicas_to_data_value(tablet_replicas));
-            this->add_custom_payload("tablet_replicas", replicas_values.serialize_nonnull());
             auto v1 = data_value(dht::token::to_int64(token_range.first));
             auto v2 = data_value(dht::token::to_int64(token_range.second));
-            bytes token_bytes(bytes::initialized_later(), v1.serialized_size() + v2.serialized_size());
-            auto i = token_bytes.begin();
-            v1.serialize(i);
-            v2.serialize(i);
-            this->add_custom_payload("token_range", token_bytes);
+
+            auto tablets_routing = make_tuple_value(tuple_type_impl::get_instance({long_type, long_type, replica::get_replica_set_type()}), {v1, v2, replicas_values});
```

Comment:
We should cache the type instance since this is on the hot path.

Reply:
It's only a hot path with a tablet-unaware driver. But yes we should cache it.

Perhaps we should only send this extra information if the driver declares support for it.